qemu/tests/balloon_service.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import time
+import os
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import error_context
+from avocado.core import exceptions
+from aexpect import ShellCmdError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Balloon service test for windows guest.
+    1) boot a windows guest with balloon device.
+    2) enable and check driver verifier in guest.
+    3) install balloon service in guest.
+    4) enable polling in qmp.
+    5) evict and enlarge balloon.
+    6) get polling value in qmp.
+    7) uninstall balloon service and clear driver verifier.
+    """
+
+    def get_disk_vol(session):
+        """
+        Get virtio-win disk volume letter.
+
+        :param session: VM session.
+
+        :return drive_letter: virtio-win disk volume letter.
+        """
+        volume_name = params["cdrom_virtiowin"]
+        volume_name = volume_name.split("/")[-1]
+        volume_name = volume_name[:volume_name.rfind(".")]
+        try:
+            return utils_misc.get_win_disk_vol(session,
+                                               key="VolumeName",
+                                               val=volume_name)
+        except ShellCmdError:
+            raise exceptions.TestFail("Could not get virtio-win disk vol!")
+
+    def config_balloon_service(session):
+        """
+        Check / Install balloon service.
+
+        :param session: VM session.
+
+        :return drive_letter: virtio-win disk volume letter.
+        """
+        drive_letter = get_disk_vol(session)
+        status_balloon_service = params["status_balloon_service"] % drive_letter
+        output = session.cmd_output(status_balloon_service)
+        if re.search(r"running", output.lower(), re.M):
+            if re.search(r"stop", output.lower(), re.M):
+                logging.debug("Install Balloon Service in guest.")
+                try:
+                    install_balloon_service = params["install_balloon_service"] % drive_letter
+                    session.cmd(install_balloon_service)
+                except ShellCmdError:
+                    raise exceptions.TestError("Install balloon service failed !")
+            else:
+                logging.debug("Run Balloon Service in guest.")
```

Comment:
Do we still need to run balloon after balloon installation,
if yes, need to remove else ...